internal/util/copy.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2022 Fraunhofer AISEC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//           $$\                           $$\ $$\   $$\
+//           $$ |                          $$ |\__|  $$ |
+//  $$$$$$$\ $$ | $$$$$$\  $$\   $$\  $$$$$$$ |$$\ $$$$$$\    $$$$$$\   $$$$$$\
+// $$  _____|$$ |$$  __$$\ $$ |  $$ |$$  __$$ |$$ |\_$$  _|  $$  __$$\ $$  __$$\
+// $$ /      $$ |$$ /  $$ |$$ |  $$ |$$ /  $$ |$$ |  $$ |    $$ /  $$ |$$ | \__|
+// $$ |      $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$ |  $$ |$$\ $$ |  $$ |$$ |
+// \$$$$$$\  $$ |\$$$$$   |\$$$$$   |\$$$$$$  |$$ |  \$$$   |\$$$$$   |$$ |
+//  \_______|\__| \______/  \______/  \_______|\__|   \____/  \______/ \__|
+//
+// This file is part of Clouditor Community Edition.
+
+package util
+
+// DeepCopy copies an interface[] array
+func DeepCopy(original []interface{}) []interface{} {
+
+	var target []interface{}
+
+	for i := range original {
+		target = append(target, original[i])
+	}
+
+	return target
+}
+
+// DeepCopyOfMap copies a map
+func DeepCopyOfMap(originalMap map[string]interface{}) map[string]interface{} {
+	targetMap := make(map[string]interface{})
+
+	for key, value := range originalMap {
+		targetMap[key] = value
```

Comment:
Wouldn't a deep copy need to check, if value is of type map and then return a deep copy of that instead of the original value?

Reply:
Probably, but in that case I don't know if it's very smart to write it our self instead of using a library.